packages/opencensus-instrumentation-http2/src/http2.ts

```diff
@@ -0,0 +1,283 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {HttpPlugin} from '@opencensus/instrumentation-http';
+import {types} from '@opencensus/opencensus-core';
+import {classes} from '@opencensus/opencensus-core';
+import {logger} from '@opencensus/opencensus-core';
+import * as http from 'http';
+import * as http2 from 'http2';
+import * as net from 'net';
+import * as shimmer from 'shimmer';
+import * as tls from 'tls';
+import * as url from 'url';
+import * as uuid from 'uuid';
+
+export type Http2Module = typeof http2;
+export type ConnectFunction = typeof http2.connect;
+type RequestFunction =
+    (this: http2.ClientHttp2Session, headers?: http2.OutgoingHttpHeaders,
+     options?: http2.ClientSessionRequestOptions) => http2.ClientHttp2Stream;
+export type CreateServerFunction = typeof http2.createServer;
+
+/** Http2 instrumentation plugin for Opencensus */
+export class Http2Plugin extends HttpPlugin {
```

Comment:
We use HttpPlugin to reuse the atributes and `traceStatus()`, which translates the status code to [trace status code](https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md#mapping-from-http-status-codes-to-trace-status-codes). We like this approach because it reuses code, but I understand your point of view. Would you prefer to remove the dependency or keep it this way?

Reply:
I agree, specially `traceStatus` and, maybe, other things that are defined in OC spec should be in `opencensus-core` if it helps on code re usability.